modules/jabba/1.0/src/R/custom_fragcounter.R

```diff
@@ -0,0 +1,1135 @@
+#' @import GenomicRanges
+#' @import gUtils
+#' @import rtracklayer
+#' @importFrom GenomeInfoDb seqlengths
+#' @importFrom data.table data.table fread rbindlist set setkey setkeyv setnames transpose as.data.table
+#' @importFrom stats cor loess predict quantile
+#' @importFrom skidb read_gencode
+#' @importFrom Biostrings alphabetFrequency
+#' @importFrom parallel mclapply
+#' @importFrom Rsamtools BamFile
+#' @importFrom grDevices col2rgb rgb png dev.off
+#' @importFrom IRanges IRanges
+#' @importFrom DNAcopy CNA segment smooth.CNA
+#' @importFrom graphics plot lines
+#' @importFrom utils globalVariables
+
+
+
+globalVariables(c(".", ".N", ".SD", ":=", "V1", "V2", "V3", "bam", "bin", "black_list_pct", "blacklisted", "child", "chr", "count", "counts", "gc.wig", "index", "ix.start", "lev0", "lev1", "map.wig", "newcount", "numlevs", "pair", "parent", "reads", "reads.corrected", "rowid", "sortSeqlevels", "tmp"))
+
+
+
+
+#' @title Multi-scale coverage correction
+#' @description Given gc and mappability coverage correction at k "nested" scales finds the coverage
+#' assignment at the finest scale that yields the best correction at every scale
+#' @name multicoco
+#' @param cov GRanges constant with GRanges of coverage samples with (by default) fields $reads, $map, $gc
+#' @param numlevs integer numbers of scales at which to correct
+#' @param base integer Scale multiplier
+#' @param fields character vector fields of gc to use as covariates
+#' @param iterative boolean whether to iterate
+#' @param presegment boolean whether to presegment
+#' @param min.segwidth integer when presegmenting, minimum segment width
+#' @param mono boolean Wether to only do single iteration at finest scale
+#' @param verbose boolean Wether to print log to console
+#' @param FUN function with which to correct coverage (by default loess
+#' correction modified from HMMcopy that takes in granges with fields
+#' $reads and other fields specified in "fields"
+#' @param ... additional args to FUN
+#' @param mc.cores integer Number of cores to use
+#' @param exome boolean If TRUE, perform correction using exons as bins instead of fixed size
+#' @author Marcin Imielinski
```

Comment:
if this is someone else's source code, can we use it directly? Or can it be modified inside the module if there are some custom edits?